libs/langgraph/langgraph/pregel/algo.py

```diff
@@ -383,31 +452,75 @@ def prepare_single_task(
     parent_ns = configurable.get(CONFIG_KEY_CHECKPOINT_NS, "")
 
     if task_path[0] == PUSH:
-        idx = int(task_path[1])
-        if idx >= len(checkpoint["pending_sends"]):
-            return
-        packet = checkpoint["pending_sends"][idx]
-        if not isinstance(packet, Send):
-            logger.warning(
-                f"Ignoring invalid packet type {type(packet)} in pending sends"
+        if len(task_path) == 2:  # TODO: remove branch in 1.0
+            # legacy SEND tasks, executed in superstep n+1
+            # (PUSH, idx of pending send)
+            idx = cast(int, task_path[1])
+            if idx >= len(checkpoint["pending_sends"]):
+                return
+            packet = checkpoint["pending_sends"][idx]
+            if not isinstance(packet, Send):
+                logger.warning(
+                    f"Ignoring invalid packet type {type(packet)} in pending sends"
+                )
+                return
+            if packet.node not in processes:
+                logger.warning(
+                    f"Ignoring unknown node name {packet.node} in pending sends"
+                )
+                return
+            # create task id
+            triggers = [PUSH]
+            checkpoint_ns = (
+                f"{parent_ns}{NS_SEP}{packet.node}" if parent_ns else packet.node
             )
+            task_id = _uuid5_str(
+                checkpoint_id,
+                checkpoint_ns,
+                str(step),
+                packet.node,
+                PUSH,
+                str(idx),
+            )
+        elif len(task_path) == 4:
+            # new PUSH tasks, executed in superstep n
+            # (PUSH, parent task path, idx of PUSH write, id of parent task)
+            task_path_t = cast(tuple[str, tuple, int, str], task_path)
+            writes_for_path = [w for w in pending_writes if w[0] == task_path_t[3]]
+            if task_path_t[2] >= len(writes_for_path):
+                logger.warning(
+                    f"Ignoring invalid write index {task_path[2]} in pending writes"
+                )
+                return
+            packet = writes_for_path[task_path_t[2]][2]
+            if not isinstance(packet, Send):
+                print("packet", task_path_t, writes_for_path)
```

Comment:
removed in https://github.com/langchain-ai/langgraph/pull/2393 which will be merged at same time